src/common/utils/PastelPromoCode.ts

```diff
@@ -0,0 +1,87 @@
+import fs from 'fs'
```

Comment:
The PastelPromoCode has been used for a few places and It can be used for other places in the future. And it is also used to check an address is a normal or promo code address too. So that's why I have put it in the util to reusable.